src/dict_mem.cpp

```diff
@@ -180,6 +173,7 @@ void DictMem::InitRootNode()
 
 DictMem::~DictMem()
 {
+    Destroy();
```

Comment:
We always call the Destroy on-demand in the code instead of using the destructor.
If you call destroy in destructor, then you need to remove the call of "mm.Destroy" in Dict::Destroy().
Or set the pointers to null after delete in DictMem::Destroy(). Otherwise, there will be double free.

Also if we want to fix this according to coverity complains, we would have to fix similar issue in the Dict destructor. To me it sounds like a false positive. However, coverity will always simply issue a memory leak issue if you don't free memory in destructor.